### PR TITLE
[JAV-397] Add automatic signing of release on tagged builds

### PR DESCRIPTION
**Pull request by asifdxtreme**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -85,19 +85,35 @@
       <id>release</id>
       <build>
         <plugins>
+        <plugin>
```
Comment (WillemJiang) [inline pom.xml]: why do we need to add the config of maven-deploy-plugin?

File: .travis.settings.xml [hunk 2]
```diff
@@ -7,4 +7,14 @@
             <password>${env.SONATYPE_PASSWORD}</password>
         </server>
     </servers>
+<profiles>
```
Comment (liubao68) [inline .travis.settings.xml]: This file is not properly indented

File: pom.xml (hunk 1 above)
Comment (asifdxtreme) [reply pom.xml]: yes it is redundant as already while calling mvn deploy we are setting skiptest so this configuration is not required, I will update the changes

File: .travis.settings.xml (hunk 2 above)
Comment (asifdxtreme) [reply .travis.settings.xml]: Thanks @liubao68 , I have updated the file with proper indentedation

File: .travis.yml [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
-  - mvn deploy -DskipTests --settings .travis.settings.xml
+ deploy:
+  provider: script
+  script: 
+   - openssl aes-256-cbc -K $encrypted_acbbc88fb3ab_key -iv $encrypted_acbbc88fb3ab_iv -in gpg-sec.tar.enc -out gpg-sec.tar -d
+   - tar xvf gpg-sec.tar
+   - mvn deploy -DskipTests -Prelease -Ppassphrase --settings .travis.settings.xml
+  on:
+   tags: true
+   all_branches: true
```
Comment (seanyinx) [inline .travis.yml]: does this mean deploying changes from all branches?

File: .travis.yml [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
 services:
-  - docker
-
+- docker
 script:
-  - mvn clean install -Ddocker.showLogs -Pdocker -Pjacoco -Pit -Pcoverage coveralls:report
-
+- mvn clean install -Ddocker.showLogs -Pdocker -Pjacoco -Pit -Pcoverage coveralls:report
 after_success:
-  - mvn deploy -DskipTests --settings .travis.settings.xml
```
Comment (seanyinx) [inline .travis.yml]: we need snapshots deployed too on each build, in order to allow testing in demo projects to discover potential bugs before release

File: .travis.yml (hunk 3 above)
Comment (asifdxtreme) [reply .travis.yml]: if there is any tagged commit from any branch then it will deploy with auto-signing

File: .travis.yml (hunk 4 above)
Comment (WillemJiang) [reply .travis.yml]: Can we just keep deploying the snapshot there?

Comment (asifdxtreme) [reply .travis.yml]: Currently I have made the changes to retain old behaviour and on tagged commit the auto signing will be performed for deploy.

File: .travis.yml [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
+- mvn deploy -DskipTests --settings .travis.settings.xml
+deploy:
+ provider: script
+ script: 
+  - openssl aes-256-cbc -K $encrypted_acbbc88fb3ab_key -iv $encrypted_acbbc88fb3ab_iv -in gpg-sec.tar.enc -out gpg-sec.tar -d
+  - tar xvf gpg-sec.tar
+  - mvn deploy -DskipTests -Prelease -Ppassphrase --settings .travis.settings.xml
+ on:
+  tags: true
+  all_branches: true
```
Comment (seanyinx) [inline .travis.yml]: is it better to limit release to tag on master only?

Comment (asifdxtreme) [reply .travis.yml]: okay, I will update it

**Pull request merged**